src/ctapipe/io/interpolation.py

```diff
@@ -0,0 +1,259 @@
+    ).tag(config=True)
+
+    table_location = None
+
+    def __init__(self, h5file=None, **kwargs):
+        super().__init__(**kwargs)
+
+        if h5file is not None and not isinstance(h5file, tables.File):
+            raise TypeError("h5file must be a tables.File")
+        self.h5file = h5file
+
+        self.interp_options: dict[str, Any] = dict(assume_sorted=True, copy=False)
+        if self.bounds_error:
+            self.interp_options["bounds_error"] = True
+        elif self.extrapolate:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = "extrapolate"
+        else:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = np.nan
+
+        self._interpolators = {}
+
+    @abstractmethod
+    def add_table(self, tel_id, input_table):
+        pass
+
+    def _check_interpolators(self, tel_id):
+        if tel_id not in self._interpolators:
+            if self.h5file is not None:
+                self._read_parameter_table(tel_id)  # might need to be removed
+            else:
+                raise KeyError(f"No table available for tel_id {tel_id}")
+
+    def _read_parameter_table(self, tel_id):
+        input_table = read_table(
+            self.h5file,
+            self.table_location + f"/tel_{tel_id:03d}",
+        )
+        print(self.table_location + f"/tel_{tel_id:03d}")
+        self.add_table(tel_id, input_table)
+
+
+class PointingInterpolator(Interpolator):
+    """
+    Interpolator for pointing and pointing correction data
+    """
+
+    table_location = "/dl0/monitoring/telescope/pointing"
+
+    def __call__(self, tel_id, time):
+        """
+        Interpolate alt/az for given time and tel_id.
+
+        Parameters
+        ----------
+        tel_id : int
+            telescope id
+        time : astropy.time.Time
+            time for which to interpolate the pointing
+
+        Returns
+        -------
+        altitude : astropy.units.Quantity[deg]
+            interpolated altitude angle
+        azimuth : astropy.units.Quantity[deg]
+            interpolated azimuth angle
+        """
+
+        self._check_interpolators(tel_id)
+
+        mjd = time.tai.mjd
+        az = u.Quantity(self._interpolators[tel_id]["az"](mjd), u.rad, copy=False)
+        alt = u.Quantity(self._interpolators[tel_id]["alt"](mjd), u.rad, copy=False)
+        return alt, az
+
+    def add_table(self, tel_id, input_table):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns for pointing and pointing correction data.
+        """
+
+        missing = {"time", "azimuth", "altitude"} - set(input_table.colnames)
```

Comment:
I think we can de-duplicate a bit of code here by adding a `required_columns` and `expected_units` argument on the classes and create a `_check_table` function on the base class.

Reply:
the calibration data will not have units, so when interpolating them we don't expect them to have a specific unit

Reply:
Sure, then just leave it empty:

```
expected_units = {}
```

the idea would be that the `_check_table` function checks for  each column in `expected_units` that the units are compatible.

You could even make sure to check that there are in fact no units, by doing:

```
expected_units = {"gain": None}
```

Reply:
I think we shall use dimensionless units instead of None. This will allow to simplify the validation a little bit too.
```
>>> from astropy.units import Unit
>>> dimensionless_unit = Unit()
>>> dimensionless_unit
Unit(dimensionless)
```